packages/dev/core/src/Cameras/arcRotateCamera.ts

```diff
@@ -899,16 +911,24 @@ export class ArcRotateCamera extends TargetCamera {
             this._currentInterpolationFactor = 0.1;
         }
 
-        alpha = Clamp(alpha, this.lowerAlphaLimit ?? -Infinity, this.upperAlphaLimit ?? Infinity);
-        beta = Clamp(beta, this.lowerBetaLimit ?? -Infinity, this.upperBetaLimit ?? Infinity);
-        radius = Clamp(radius, this.lowerRadiusLimit ?? -Infinity, this.upperRadiusLimit ?? Infinity);
-        target.y = Clamp(target.y, this.lowerTargetYLimit ?? -Infinity, Infinity);
+        // If NaN is passed in for a goal value, keep the current goal value.
+        const selectGoalValue = (newGoal: number, currentGoal: number): number => (isNaN(newGoal) ? currentGoal : newGoal);

```

Comment:
maybe move it outside the loop as we do not need to redefined it everytime

Reply:
Ok I changed it to use a helper function at the module level. That said, from what I've observed, modern JS engines are smart enough to not re-allocate lambdas that don't capture local state. It's true though that making it a module level function eliminates a dependency on this JS engine implementation detail.

Reply:
oh interesting! how do you see that?

Reply:
It came up previously with Viewer related stuff and I investigated a bit. You can easily see it with something like this:
```ts
function outer() {
    const someLocal = performance.now();

    const lambdaWithCapture = () => {
        console.log(`With capture: ${someLocal}`);
    };
    lambdaWithCapture();

    const lambdaWithoutCapture = () => {
        console.log("Without capture");
    };
    lambdaWithoutCapture();
}

for (let i = 0; i < 100; i++) {
    outer();
}
```

If you do a memory profile, and look at allocations, you will see:
![image](https://github.com/user-attachments/assets/819d5c06-7121-4cb3-a752-140cf00e78e5)

Admittedly I don't know why the count is 201 instead of 100. But in comparison the `lambdaWithoutCapture` is just 1.

Reply:
excellent! thanks for the good learning!